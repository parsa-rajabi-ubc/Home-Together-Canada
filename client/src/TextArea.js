import React from 'react';

export function TextArea(props){
    return(
        <label>
            {props.label}
            <input type="text" placeholder={props.placeholder} />
        </label>
    );
}