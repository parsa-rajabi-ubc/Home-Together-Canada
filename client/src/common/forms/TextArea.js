import React from 'react';
import PropTypes from "prop-types";
/**
 * @Author:     Jeff Hatton
 * @Created:    2020.11.05
 *
 * @Description: Test box input Form Component
 *
 */

//returns generic textbox field for a form with a label
function TextArea(props){
    const { onChange } = props;

    return(
        <label>
            {props.label}
            <input type="text" onChange= {onChange}/>
        </label>
    );
}

TextArea.propTypes = {
    label: PropTypes.string.isRequired,
    onChange: PropTypes.func
}

export default TextArea;
