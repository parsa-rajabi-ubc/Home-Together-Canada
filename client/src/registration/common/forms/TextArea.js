import React from 'react';
import PropTypes from "prop-types";

function TextArea(props){
    return(
        <label>
            {props.label}
            <input type="text" placeholder={props.placeholder} />
        </label>
    );
}
TextArea.propTypes = {
    label: PropTypes.string.isRequired,
    placeholder: PropTypes.string.isRequired
}

export default TextArea