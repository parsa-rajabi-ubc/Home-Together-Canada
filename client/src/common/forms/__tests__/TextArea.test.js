import React from 'react';
import {TextArea} from "../TextArea";
import renderer from  'react-test-renderer'

describe('TextArea', () => {
    describe('Snapshot test', () => {

        it("should render correctly regardless of properties", () => {

            //when
                const component1 = renderer.create(<TextArea label="testLabelString1" placeholder="testPlaceholderString1"/>).toJSON();
            //then
            expect(component1).toMatchSnapshot();

            //when
                const component2 = renderer.create(<TextArea label="testLabelString2" placeholder="testPlaceholderString2"/>).toJSON();
            //then
            expect(component2).toMatchSnapshot();
        });
    })
})