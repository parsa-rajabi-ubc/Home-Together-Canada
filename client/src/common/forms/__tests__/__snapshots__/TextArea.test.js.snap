// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea Snapshot test should render correctly regardless of properties 1`] = `
<label>
  testLabelString1
  
  <input
    className="testClassNameString"
    disabled={false}
    placeholder="testPlaceholderString2"
    type="text"
  />
</label>
`;
